ledger/src/blockstore.rs

```diff
@@ -11837,7 +11839,7 @@ pub mod tests {
         slot_meta.last_index = Some(last_index as u64);
         blockstore.put_meta(slot, &slot_meta).unwrap();
         let results = blockstore.check_last_fec_set(slot).unwrap();
-        assert!(results.last_fec_set_merkle_root.is_none());
+        assert_eq!(results.last_fec_set_merkle_root, Some(merkle_root));
```

Comment:
Why does the merkle root exist now where it did not exist before?

Reply:
Old code:
1. Last shred index is 60
2. We check indices 29..60
3. These indices span across 2 different FEC sets
4. We read up the shreds and extract merkles
5. We dedup the merkles
6. We detect we have 2 unique merkles after deduping and return `None` as `last_fec_set_merkle_root`

New code:
1. Last shred index is 63
2. We check indices 32..63
3. These indices span across 1 FEC set
4. We read up the shreds and extract merkles
5. We dedup the merkles
6. We only have 1 unique Merkle and return `Some(<hash>)` as `last_fec_set_merkle_root`

Reply:
I do not understand. we made 2 fec sets initially, we still make 2 fec sets. granted they are different length now, but the rest of the logic is the same. we are not messing with indices directly, so both sets should be intact. why did the story of unique merkles change here?

Reply:
Discussed elsewhere:
The test always looks at the range from last_index-31..last_index. Current master is only generating 30 shreds for the final FEC set. Thus, this hard coded -31 causes us to span 2 FEC sets, and thus we see 2 unique Merkles.

The reason the test has a hard coded -31 is because the final FEC set is supposed to be padded out to 32:32 shreds at least.

The reason it does not get padded out to 32:32 in the test is because we don’t actually pass in the “is last” flag or whatever when generating the shreds. We just manually manipulate the last index flag.